File: Watch/app/interface_controller.rb
class InterfaceController < WKInterfaceController
  extend IB

  outlet :periodNameLabel, WKInterfaceLabel
  outlet :clockImage, WKInterfaceImage
  outlet :timeRemainingLabel, WKInterfaceLabel

  def initWithContext(context)
    super

    if WKInterfaceDevice.currentDevice.screenBounds.size.width < 156 
      @clockRect = CGRectMake(0, 0, 100, 100)
    else
      @clockRect = CGRectMake(0, 0, 120, 120)
    end

    return self
  end

  def willActivate
    FkP.setup
    @tick = NSTimer.scheduledTimerWithTimeInterval(1, target: self, selector: "refresh", userInfo: nil, repeats: true) if @tick.nil?
    refresh
  end

  def didDeactivate
    unless @tick.nil?
      @tick.invalidate
      @tick = nil
    end
  end

  def refresh
    status = FkP.status(@clockRect)
    @clockImage.image = status[:clock]
    @periodNameLabel.text = status[:periodName]
    @timeRemainingLabel.text = status[:timeRemaining]
  end

end

File: Watch App Extension/InterfaceController.swift
import WatchKit
import Foundation
import KingPastaKitWatch

class InterfaceController: WKInterfaceController {
    
    var clockRect = WKInterfaceDevice.currentDevice().screenBounds.width < 156 ? CGRect(x: 0, y: 0, width: 100, height: 100) : CGRect(x: 0, y: 0, width: 120, height: 120)
    
    private var tick: NSTimer?
    
    @IBOutlet weak var periodNameLabel: WKInterfaceLabel!
    @IBOutlet weak var clockImage: WKInterfaceImage!
    @IBOutlet weak var timeRemainingLabel: WKInterfaceLabel!

    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
        //
    }

    override func willActivate() {
        super.willActivate()
        
        if FkP.isInitialSetupComplete {
            if tick == nil {
                tick = NSTimer.scheduledTimerWithTimeInterval(1, target: self, selector: "refresh", userInfo: nil, repeats: true)
                refresh()
            }
        } else {
            clockImage.setImage(Clock.blank(clockRect))
            periodNameLabel.setText("Free time")
            timeRemainingLabel.setText("Nothing scheduled")
        }
    }

    override func didDeactivate() {
        super.didDeactivate()
        
        if let tick = tick {
            tick.invalidate()
            self.tick = nil
        }
    }
    
    func refresh() {
        let status = FkP.status(clockRect)
        clockImage.setImage(status.clock)
        periodNameLabel.setText(status.periodName)
        timeRemainingLabel.setText(status.timeRemaining)
    }

}
